googleapiclient/discovery.py

```diff
@@ -212,10 +255,9 @@ def build(serviceName,
   except ValueError as e:
     logger.error('Failed to parse as JSON: ' + content)
     raise InvalidJsonError()
-
-  return build_from_document(content, base=discoveryServiceUrl, http=http,
-      developerKey=developerKey, model=model, requestBuilder=requestBuilder,
-      credentials=credentials)
+  if cache_discovery and cache and isinstance(cache, base.CacheBase):
```

Comment:
"if cache_discovery and cache is not None:" - again, no need to typecheck here, unless I'm missing something.